lib/internal/fs/watchers.js

```diff
@@ -367,10 +389,23 @@ async function* watch(filename, options = kEmptyObject) {
     }
 
     while (!signal?.aborted) {
-      yield await promise;
-      ({ promise, resolve, reject } = PromiseWithResolvers());
+      await promise;
+      while (queue.length) {
+        const item = ArrayPrototypeShift(queue);
+        if (
+          ObjectPrototypeHasOwnProperty(item, 'eventType') &&
+          ObjectPrototypeHasOwnProperty(item, 'filename')
+        ) {
+          yield item;
+        } else {
+          throw item;
```

Comment:
```suggestion
        if (item instanceof Error) {
          throw item;
        } else {
          yield item;
```

I think this should be enough - would've been better if we have `Error.isError` but then I don't think it's possible for things from another realm to end up here.